### PR TITLE
Replace Gatsby plugins with new built-in functionalities from Gatsby

### LINKED ISSUE #1389: Use Gatsby's built-in support for modifying the document head
```
warning gatsby-plugin-react-helmet: Gatsby now has built-in support for modifying the document head.
```

Learn more at https://gatsby.dev/gatsby-head

### LINKED ISSUE #1390: Use Gatsby's `trailingSlash` option
```
warning gatsby-plugin-remove-trailing-slashes: Gatsby now has a trailingSlash option.
```

Learn more at https://gatsby.dev/trailing-slash

### LINKED ISSUE #1456: Incorrect page title in static pregenerate HTML (only react hydration sets the correct title)
The commercetools Frontend Documentation page shows: "commercetools Composable Commerce documentation" as the title before React Hydration (JavaScript execution) and replaced with the correct title after hydration.

The static pre-generated site seems to have the "commercetools Composable Commerce documentation" as the title, only replaced by the correct one after React Hydration. It affects all open graph `<meta>` tags.

The image below is the Slack preview of the URL: https://docs.commercetools.com/docs/frontend.

08-12-2022-at-15-54-23@2x

**Recreating the issue**

Use the https://docs.commercetools.com/docs/frontend link in a Slack message and check the link preview.

or

Disable JavaScript in your browser and open the  [URL](https://docs.commercetools.com/docs/frontend link) in the browser

08-12-2022-at-16-07-12@2x

### PR DESCRIPTION
**Pull request by timonrey**
Replaces the `gatsby-plugin-react-helmet` plugin with Gatsby head
https://www.gatsbyjs.com/docs/reference/built-in-components/gatsby-head/

Replaces the `gatsby-plugin-remove-trailing-slashes`plugin with Gatsby trailing slash
https://www.gatsbyjs.com/docs/reference/config-files/gatsby-config/#trailingslash

closes #1390 
closes #1389 
closes #1456

### CODE REVIEW

File: packages/gatsby-theme-docs/gatsby-ssr.js [hunk 1]
```diff
@@ -56,6 +57,7 @@ export const replaceRenderer = ({
     .replace(lobotomizedOwlSelectorRegex, patchedLobotomizedOwlSelector);
 
   replaceBodyHTMLString(patchedHtml);
+  setHtmlAttributes({ lang: 'en' });
```
Comment (timonrey) [inline packages/gatsby-theme-docs/gatsby-ssr.js:60]: Part of the react-helmet to gatsby head migration

File: packages/gatsby-theme-docs/src/templates/homepage.js [hunk 2]
```diff
@@ -66,6 +58,21 @@ HomepageTemplate.propTypes = {
 };
 export default HomepageTemplate;
 
+// eslint-disable-next-line react/prop-types
```
Comment (timonrey) [inline packages/gatsby-theme-docs/src/templates/homepage.js:61]: It looks like there is no way to define PropTypes for the Head function. So I needed to disable eslint for the lines where we define the props for the SEO component.

File: packages/ui-kit/src/components/mermaid-client-side.tsx [hunk 3]
```diff
@@ -126,7 +126,12 @@ const Mermaid = ({ graph }: MermaidProps) => {
     });
   }, [graph]);
 
-  return <Figure dangerouslySetInnerHTML={{ __html: svg }}></Figure>;
+  return (
+    <Figure
+      data-test-id="mermaid-diagram"
```
Comment (timonrey) [inline packages/ui-kit/src/components/mermaid-client-side.tsx:131]: To test the js availability in the e2e test.

**Review by gabriele-ct — APPROVED** [review]

File: packages/gatsby-theme-docs/gatsby-config.js [hunk 4]
```diff
@@ -306,9 +297,8 @@ module.exports = (themeOptions = {}) => {
       },
 
       /**
-       * The following plugins need to be last
+       * The following plugin need to be last
        */
-      'gatsby-plugin-remove-trailing-slashes',
       'gatsby-plugin-meta-redirect',
```
Comment (nkuehn) [inline packages/gatsby-theme-docs/gatsby-config.js:302]: @timonrey  i know it's merged now but as far as I know the meta redirect plugin can also be removed now.  We're not using the function anywhere - at least I was not able to find it.

**Pull request merged**